### ISSUE TITLE
piolib_mod.F90 init_intercom completely broken

### ISSUE THREAD

**Issue by jedwards4b** [NCAR/ParallelIO#1251]
In the rewrite of pio_init_async in c we commented out the fortran interface and have not replaced it.

Why does pio_init_async insist on creating comms internally?   Many models including cesm have very sophisticated routines for defining comms.  Can we create a pio_init_async interface which accepts and dups comm from the model?

**Comment by edhartnett** [comment]
Yes we can add a new init_async that takes the comms already defined. I will do so...

**Comment by edhartnett** [comment]
I'm back on PIO this week and will address this issue...

**Comment by jedwards4b** [comment]
Have you had a chance to look at this one yet?

**Comment by edhartnett** [comment]
No, sorry, but it is coming.

**Comment by jedwards4b** [comment]
@edhartnett can you take another look at this issue?

**Comment by edhartnett** [comment]
No because I cannot use the most recent netCDF without this issue being resolved: #1070.

This is happening in generated fortran test code, making it hard to debug. It has nothing to do with async code, but I don't know what is causing it. I have a C test that seems to reproduce the same exact behavior without hanging.

The problem is somewhere in the code relating to collective access. Somewhere one of the PIO tasks is not calling nc_put_vara() as it should, and so everything is hanging. (This is my current theory.)

I thought this was because of a bug in netCDF, so I added a bunch of parallel netCDF testing, but it all seems to be working just fine. So I think the problem is in the PIO library. If you can fix it I will not only turn immediately to this issue but will also buy you a beer if that is allowed by government regulations. (Which, I suppose, it is not.)

**Comment by edhartnett** [comment]
I have been tracing through the code and I get lost in the rearrangers. In the fortran test, is there any reason why any of the tasks should ever have a count of 0? Seems like they should all have counts above 0 at all times. But sometimes in the log it is showing counts of 0....
